local Arsenic = {
    Name = "Arsenic",
    Symbol = "As",
    AtomicNumber = 33,
    AtomicMass = 74.922,
    RepresentativeMassNumber = 75,
    Protons = 33,
    Electrons = 33,
    NeutronsApprox = 42,
    Category = "Metalloid",
    Group = "15",
    Period = 4,
    Block = "p",
    StateAt293K = "Solid",
    Density = 5.727,
    Appearance = "Arsenic is a semi-metal. In its metallic form it is bright, silver-grey and brittle.",
    Uses = {
        "Arsenic is a well-known poison. Arsenic compounds are sometimes used as rat poisons and insecticides but their use is strictly controlled.",
        "Surprisingly, arsenic can also have medicinal applications. In Victorian times, Dr Fowler's Solution (potassium arsenate dissolved in water) was a popular cure-all tonic that was even used by Charles Dickens. Today, organoarsenic compounds are added to poultry feed to prevent disease and improve weight gain.",
        "Arsenic is used as a doping agent in semiconductors (gallium arsenide) for solid-state devices. It is also used in bronzing, pyrotechnics and for hardening shot.",
        "Arsenic compounds can be used to make special glass and preserve wood.",
    },
    ElectronConfiguration = "[Ar] 3d104s24p3",
    EnergyLevels = {
        2,
        8,
        18,
        5,
    },
    AtomicRadius = 1.85,
    CovalentRadius = 1.2,
    ElectronAffinity = 77.574,
    Electronegativity = 2.18,
    IonizationEnergies = {
        944.456,
        1793.585,
        2735.456,
        4836.81,
        6042.88,
        12311.5,
    },
    MeltingPointK = 889,
    KeyIsotopes = {
        {
            Name = "75As",
            HalfLife = {},
            Mass = 74.922,
            MassNumber = 75,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
    Isotopes = {
        {
            Name = "75As",
            HalfLife = {},
            Mass = 74.922,
            MassNumber = 75,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
}

return Arsenic
