local Beryllium = {
    Name = "Beryllium",
    Symbol = "Be",
    AtomicNumber = 4,
    AtomicMass = 9.0122,
    RepresentativeMassNumber = 9,
    Protons = 4,
    Electrons = 4,
    NeutronsApprox = 5,
    Category = "Alkaline",
    Group = "2",
    Period = 2,
    Block = "s",
    StateAt293K = "Solid",
    Density = 1.85,
    Appearance = "Beryllium is a silvery-white metal. It is relatively soft and has a low density.",
    Uses = {
        "Beryllium is used in alloys with copper or nickel to make gyroscopes, springs, electrical contacts, spot-welding electrodes and non-sparking tools. Mixing beryllium with these metals increases their electrical and thermal conductivity.",
        "Other beryllium alloys are used as structural materials for high-speed aircraft, missiles, spacecraft and communication satellites.",
        "Beryllium is relatively transparent to X-rays so ultra-thin beryllium foil is finding use in X-ray lithography. Beryllium is also used in nuclear reactors as a reflector or moderator of neutrons.",
        "The oxide has a very high melting point making it useful in nuclear work as well as having ceramic applications.",
    },
    ElectronConfiguration = "[He] 2s2",
    EnergyLevels = {
        2,
        2,
    },
    AtomicRadius = 1.53,
    CovalentRadius = 0.99,
    Electronegativity = 1.57,
    IonizationEnergies = {
        899.504,
        1757.108,
        14848.767,
        21006.658,
    },
    MeltingPointK = 1560,
    BoilingPointK = 2741,
    KeyIsotopes = {
        {
            Name = "9Be",
            HalfLife = {},
            Mass = 9.012,
            MassNumber = 9,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
    Isotopes = {
        {
            Name = "9Be",
            HalfLife = {},
            Mass = 9.012,
            MassNumber = 9,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
}

return Beryllium
