local Dysprosium = {
    Appearance = "A bright, silvery metallic element.",
    AtomicMass = 162.5,
    AtomicNumber = 66,
    AtomicRadius = 2.31,
    Block = "f",
    BoilingPointK = 2840,
    Category = "Lanthanoid",
    CovalentRadius = 1.8,
    ElectronConfiguration = "[Xe] 4f106s2",
    Electronegativity = 1.22,
    Density = 8.54,
    Electrons = 66,
    EnergyLevels = {
        2,
        8,
        18,
        28,
        8,
        2,
    },
    Group = "Lanthanides",
    IonizationEnergies = {
        573.017,
        1125.98,
        2199.9,
        4001.25,
    },
    Isotopes = {
        {
            HalfLife = {},
            Mass = 155.924,
            MassNumber = 156,
            ModeOfDecay = {},
            Name = "156Dy",
            NaturalAbundance = 0.056,
        },
        {
            HalfLife = {},
            Mass = 157.924,
            MassNumber = 158,
            ModeOfDecay = {},
            Name = "158Dy",
            NaturalAbundance = 0.095,
        },
        {
            HalfLife = {},
            Mass = 159.925,
            MassNumber = 160,
            ModeOfDecay = {},
            Name = "160Dy",
            NaturalAbundance = 2.329,
        },
        {
            HalfLife = {},
            Mass = 160.927,
            MassNumber = 161,
            ModeOfDecay = {},
            Name = "161Dy",
            NaturalAbundance = 18.889,
        },
        {
            HalfLife = {},
            Mass = 161.927,
            MassNumber = 162,
            ModeOfDecay = {},
            Name = "162Dy",
            NaturalAbundance = 25.475,
        },
        {
            HalfLife = {},
            Mass = 162.929,
            MassNumber = 163,
            ModeOfDecay = {},
            Name = "163Dy",
            NaturalAbundance = 24.896,
        },
        {
            HalfLife = {},
            Mass = 163.929,
            MassNumber = 164,
            ModeOfDecay = {},
            Name = "164Dy",
            NaturalAbundance = 28.26,
        },
    },
    KeyIsotopes = {
        {
            HalfLife = {},
            Mass = 163.929,
            MassNumber = 164,
            ModeOfDecay = {},
            Name = "164Dy",
            NaturalAbundance = 28.26,
        },
    },
    MeltingPointK = 1685,
    Name = "Dysprosium",
    NeutronsApprox = 98,
    Period = 6,
    Protons = 66,
    RepresentativeMassNumber = 164,
    StateAt293K = "Solid",
    Symbol = "Dy",
    Uses = "As a pure metal it is little used, because it reacts readily with water and air. Dysprosium's main use is in alloys for neodymium-based magnets. This is because it is resistant to demagnetisation at high temperatures. This property is important for magnets used in motors or generators. These magnets are used in wind turbines and electrical vehicles, so demand for dysprosium is growing rapidly. \
\
Dysprosium iodide is used in halide discharge lamps. The salt enables the lamps to give out a very intense white light.\
\
A dysprosium oxide-nickel cermet (a composite material of ceramic and metal) is used in nuclear reactor control rods. It readily absorbs neutrons, and does not swell or contract when bombarded with neutrons for long periods.",
}

return Dysprosium
