local Erbium = {
    Name = "Erbium",
    Symbol = "Er",
    AtomicNumber = 68,
    AtomicMass = 167.26,
    RepresentativeMassNumber = 166,
    Protons = 68,
    Electrons = 68,
    NeutronsApprox = 98,
    Category = "Lanthanoid",
    Group = "Lanthanides",
    Period = 6,
    Block = "f",
    StateAt293K = "Solid",
    Density = 9.066,
    Appearance = "A soft, silvery metallic element.",
    Uses = {
        "Erbium finds little use as a metal because it slowly tarnishes in air and is attacked by water.",
        "When alloyed with metals such as vanadium, erbium lowers their hardness and improves their workability.",
        "Erbium oxide is occasionally used in infrared absorbing glass, for example safety glasses for welders and metal workers. When erbium is added to glass it gives the glass a pink tinge. It is used to give colour to some sunglasses and imitation gems.",
        "Broadband signals, carried by fibre optic cables, are amplified by including erbium in the glass fibre.",
    },
    ElectronConfiguration = "[Xe] 4f126s2",
    EnergyLevels = {
        2,
        8,
        18,
        30,
        8,
        2,
    },
    AtomicRadius = 2.29,
    CovalentRadius = 1.77,
    Electronegativity = 1.24,
    IonizationEnergies = {
        589.304,
        1151.07,
        2194.08,
        4119.9,
    },
    MeltingPointK = 1802,
    BoilingPointK = 3141,
    KeyIsotopes = {
        {
            Name = "166Er",
            HalfLife = {},
            Mass = 165.93,
            MassNumber = 166,
            ModeOfDecay = {},
            NaturalAbundance = 33.503,
        },
    },
    Isotopes = {
        {
            Name = "162Er",
            HalfLife = {},
            Mass = 161.929,
            MassNumber = 162,
            ModeOfDecay = {},
            NaturalAbundance = 0.139,
        },
        {
            Name = "164Er",
            HalfLife = {},
            Mass = 163.929,
            MassNumber = 164,
            ModeOfDecay = {},
            NaturalAbundance = 1.601,
        },
        {
            Name = "166Er",
            HalfLife = {},
            Mass = 165.93,
            MassNumber = 166,
            ModeOfDecay = {},
            NaturalAbundance = 33.503,
        },
        {
            Name = "167Er",
            HalfLife = {},
            Mass = 166.932,
            MassNumber = 167,
            ModeOfDecay = {},
            NaturalAbundance = 22.869,
        },
        {
            Name = "168Er",
            HalfLife = {},
            Mass = 167.932,
            MassNumber = 168,
            ModeOfDecay = {},
            NaturalAbundance = 26.978,
        },
        {
            Name = "170Er",
            HalfLife = {},
            Mass = 169.935,
            MassNumber = 170,
            ModeOfDecay = {},
            NaturalAbundance = 14.91,
        },
    },
}

return Erbium
