local Iridium = {
    Appearance = "Iridium is a hard, silvery metal. It is almost as unreactive as gold. It has a very high density and melting point.",
    AtomicMass = 192.22,
    AtomicNumber = 77,
    AtomicRadius = 2.13,
    Block = "d",
    BoilingPointK = 4701,
    Category = "Transition",
    CovalentRadius = 1.32,
    ElectronAffinity = 150.884,
    ElectronConfiguration = "[Xe] 4f145d76s2",
    Electronegativity = 2.2,
    Density = 22.56,
    Electrons = 77,
    EnergyLevels = {
        2,
        8,
        18,
        32,
        15,
        2,
    },
    Group = "9",
    IonizationEnergies = {
        865.186,
    },
    Isotopes = {
        {
            HalfLife = {},
            Mass = 190.961,
            MassNumber = 191,
            ModeOfDecay = {},
            Name = "191Ir",
            NaturalAbundance = 37.3,
        },
        {
            HalfLife = {},
            Mass = 192.963,
            MassNumber = 193,
            ModeOfDecay = {},
            Name = "193Ir",
            NaturalAbundance = 62.7,
        },
    },
    KeyIsotopes = {
        {
            HalfLife = {},
            Mass = 192.963,
            MassNumber = 193,
            ModeOfDecay = {},
            Name = "193Ir",
            NaturalAbundance = 62.7,
        },
    },
    MeltingPointK = 2719,
    Name = "Iridium",
    NeutronsApprox = 116,
    Period = 6,
    Protons = 77,
    RepresentativeMassNumber = 193,
    StateAt293K = "Solid",
    Symbol = "Ir",
    Uses = "Iridium is the most corrosion-resistant material known. It is used in special alloys and forms an alloy with osmium, which is used for pen tips and compass bearings. It was used in making the standard metre bar, which is an alloy of 90% platinum and 10% iridium. It is also used for the contacts in spark plugs because of its high melting point and low reactivity.",
}

return Iridium
