local Lutetium = {
    Name = "Lutetium",
    Symbol = "Lu",
    AtomicNumber = 71,
    AtomicMass = 174.97,
    RepresentativeMassNumber = 175,
    Protons = 71,
    Electrons = 71,
    NeutronsApprox = 104,
    Category = "Lanthanoid",
    Group = "Lanthanides",
    Period = 6,
    Block = "f",
    StateAt293K = "Solid",
    Density = 9.841,
    Appearance = "A silvery-white, hard, dense metal.",
    Uses = {
        "Lutetium is little used outside research. One of its few commercial uses is as a catalyst for cracking hydrocarbons in oil refineries.",
    },
    ElectronConfiguration = "[Xe] 4f145d16s2",
    EnergyLevels = {
        2,
        8,
        18,
        32,
        9,
        2,
    },
    AtomicRadius = 2.24,
    CovalentRadius = 1.74,
    ElectronAffinity = 32.81,
    Electronegativity = 1,
    IonizationEnergies = {
        523.516,
        1341.1,
        2022.275,
        4365.96,
        6445.2,
    },
    MeltingPointK = 1936,
    BoilingPointK = 3675,
    KeyIsotopes = {
        {
            Name = "175Lu",
            HalfLife = {},
            Mass = 174.941,
            MassNumber = 175,
            ModeOfDecay = {},
            NaturalAbundance = 97.4,
        },
    },
    Isotopes = {
        {
            Name = "175Lu",
            HalfLife = {},
            Mass = 174.941,
            MassNumber = 175,
            ModeOfDecay = {},
            NaturalAbundance = 97.4,
        },
        {
            Name = "176Lu",
            HalfLife = {
                "3.73 x 1010 y",
            },
            Mass = 175.943,
            MassNumber = 176,
            ModeOfDecay = {
                "beta-",
                "beta+",
                "EC",
            },
            NaturalAbundance = 2.6,
        },
    },
}

return Lutetium
