local Molybdenum = {
    Appearance = "A shiny, silvery metal.",
    AtomicMass = 95.95,
    AtomicNumber = 42,
    AtomicRadius = 2.17,
    Block = "d",
    BoilingPointK = 4912,
    Category = "Transition",
    CovalentRadius = 1.46,
    ElectronAffinity = 72.171,
    ElectronConfiguration = "[Kr] 4d55s1",
    Electronegativity = 2.16,
    Density = 10.28,
    Electrons = 42,
    EnergyLevels = {
        2,
        8,
        18,
        13,
        1,
    },
    Group = "6",
    IonizationEnergies = {
        684.316,
        1559.2,
        2617.65,
        4476.9,
        5257.49,
        6640.854,
        12124.734,
        13855.3,
    },
    Isotopes = {
        {
            HalfLife = {
                "> 3 x 1017 y",
            },
            Mass = 91.907,
            MassNumber = 92,
            ModeOfDecay = {
                "beta+-EC",
            },
            Name = "92Mo",
            NaturalAbundance = 14.53,
        },
        {
            HalfLife = {},
            Mass = 93.905,
            MassNumber = 94,
            ModeOfDecay = {},
            Name = "94Mo",
            NaturalAbundance = 9.15,
        },
        {
            HalfLife = {},
            Mass = 94.906,
            MassNumber = 95,
            ModeOfDecay = {},
            Name = "95Mo",
            NaturalAbundance = 15.8,
        },
        {
            HalfLife = {},
            Mass = 95.905,
            MassNumber = 96,
            ModeOfDecay = {},
            Name = "96Mo",
            NaturalAbundance = 16.67,
        },
        {
            HalfLife = {},
            Mass = 96.906,
            MassNumber = 97,
            ModeOfDecay = {},
            Name = "97Mo",
            NaturalAbundance = 9.6,
        },
        {
            HalfLife = {},
            Mass = 97.905,
            MassNumber = 98,
            ModeOfDecay = {},
            Name = "98Mo",
            NaturalAbundance = 24.39,
        },
        {
            HalfLife = {
                "6 x 1020 y",
            },
            Mass = 99.907,
            MassNumber = 100,
            ModeOfDecay = {
                "beta-beta-",
            },
            Name = "100Mo",
            NaturalAbundance = 9.82,
        },
    },
    KeyIsotopes = {
        {
            HalfLife = {},
            Mass = 94.906,
            MassNumber = 95,
            ModeOfDecay = {},
            Name = "95Mo",
            NaturalAbundance = 15.8,
        },
        {
            HalfLife = {},
            Mass = 95.905,
            MassNumber = 96,
            ModeOfDecay = {},
            Name = "96Mo",
            NaturalAbundance = 16.67,
        },
        {
            HalfLife = {},
            Mass = 97.905,
            MassNumber = 98,
            ModeOfDecay = {},
            Name = "98Mo",
            NaturalAbundance = 24.39,
        },
    },
    MeltingPointK = 2895,
    Name = "Molybdenum",
    NeutronsApprox = 53,
    Period = 5,
    Protons = 42,
    RepresentativeMassNumber = 95,
    StateAt293K = "Solid",
    Symbol = "Mo",
    Uses = "Molybdenum has a very high melting point so it is produced and sold as a grey powder. Many molybdenum items are formed by compressing the powder at a very high pressure. \
\
Most molybdenum is used to make alloys. It is used in steel alloys to increase strength, hardness, electrical conductivity and resistance to corrosion and wear. These 'moly steel' alloys are used in parts of engines. Other alloys are used in heating elements, drills and saw blades. \
\
Molybdenum disulfide is used as a lubricant additive. Other uses for molybdenum include catalysts for the petroleum industry, inks for circuit boards, pigments and electrodes.",
}

return Molybdenum
