local Praseodymium = {
    Name = "Praseodymium",
    Symbol = "Pr",
    AtomicNumber = 59,
    AtomicMass = 140.91,
    RepresentativeMassNumber = 141,
    Protons = 59,
    Electrons = 59,
    NeutronsApprox = 82,
    Category = "Lanthanoid",
    Group = "Lanthanides",
    Period = 6,
    Block = "f",
    StateAt293K = "Solid",
    Density = 6.77,
    Appearance = "A soft, silvery metal.",
    Uses = {
        "Praseodymium is used in a variety of alloys. The high-strength alloy it forms with magnesium is used in aircraft engines. Mischmetal is an alloy containing about 5% praseodymium and is used to make flints for cigarette lighters. Praseodymium is also used in alloys for permanent magnets.",
        "Along with other lanthanide elements, it is used in carbon arc electrodes for studio lighting and projection.",
        "Praseodymium salts are used to colour glasses, enamel and glazes an intense and unusually clean yellow. Praseodymium oxide is a component of didymium glass (along with neodymium). This glass is used in goggles used by welders and glassmakers, because it filters out the yellow light and infrared (heat) radiation.",
    },
    ElectronConfiguration = "[Xe] 4f36s2",
    EnergyLevels = {
        2,
        8,
        18,
        21,
        8,
        2,
    },
    AtomicRadius = 2.4,
    CovalentRadius = 1.9,
    ElectronAffinity = 92.819,
    Electronegativity = 1.13,
    IonizationEnergies = {
        528.064,
        1017.92,
        2086.399,
        3761,
        5550.8,
    },
    MeltingPointK = 1204,
    BoilingPointK = 3793,
    KeyIsotopes = {
        {
            Name = "141Pr",
            HalfLife = {},
            Mass = 140.908,
            MassNumber = 141,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
    Isotopes = {
        {
            Name = "141Pr",
            HalfLife = {},
            Mass = 140.908,
            MassNumber = 141,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
}

return Praseodymium
