local Rhodium = {
    Name = "Rhodium",
    Symbol = "Rh",
    AtomicNumber = 45,
    AtomicMass = 102.91,
    RepresentativeMassNumber = 103,
    Protons = 45,
    Electrons = 45,
    NeutronsApprox = 58,
    Category = "Transition",
    Group = "9",
    Period = 5,
    Block = "d",
    StateAt293K = "Solid",
    Density = 12.41,
    Appearance = {
        "A hard, shiny, silvery metal.",
    },
    Uses = {
        "The major use of rhodium is in catalytic converters for cars (80%). It reduces nitrogen oxides in exhaust gases.",
        "Rhodium is also used as catalysts in the chemical industry, for making nitric acid, acetic acid and hydrogenation reactions.",
        "It is used to coat optic fibres and optical mirrors, and for crucibles, thermocouple elements and headlight reflectors. It is used as an electrical contact material as it has a low electrical resistance and is highly resistant to corrosion.",
    },
    ElectronConfiguration = "[Kr] 4d85s1",
    EnergyLevels = {
        2,
        8,
        18,
        16,
        1,
    },
    AtomicRadius = 2.1,
    CovalentRadius = 1.34,
    ElectronAffinity = 109.704,
    Electronegativity = 2.28,
    IonizationEnergies = {
        719.675,
        1744.45,
        2996.83,
    },
    MeltingPointK = 2236,
    BoilingPointK = 3968,
    KeyIsotopes = {
        {
            Name = "103Rh",
            HalfLife = {},
            Mass = 102.905,
            MassNumber = 103,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
    Isotopes = {
        {
            Name = "103Rh",
            HalfLife = {},
            Mass = 102.905,
            MassNumber = 103,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
}

return Rhodium
