local Ruthenium = {
    Appearance = "A shiny, silvery metal.",
    AtomicMass = 101.07,
    AtomicNumber = 44,
    AtomicRadius = 2.13,
    Block = "d",
    BoilingPointK = 4420,
    Category = "Transition",
    CovalentRadius = 1.36,
    ElectronAffinity = 101.31,
    ElectronConfiguration = "[Kr] 4d75s1",
    Electronegativity = 2.2,
    Density = 12.45,
    Electrons = 44,
    EnergyLevels = {
        2,
        8,
        18,
        15,
        1,
    },
    Group = "8",
    IonizationEnergies = {
        710.18,
        1617.09,
        2746.94,
    },
    Isotopes = {
        {
            HalfLife = {
                "> 3.1 x 1016 y",
            },
            Mass = 95.908,
            MassNumber = 96,
            ModeOfDecay = {
                "beta+beta+",
            },
            Name = "96Ru",
            NaturalAbundance = 5.54,
        },
        {
            HalfLife = {},
            Mass = 97.905,
            MassNumber = 98,
            ModeOfDecay = {},
            Name = "98Ru",
            NaturalAbundance = 1.87,
        },
        {
            HalfLife = {},
            Mass = 98.906,
            MassNumber = 99,
            ModeOfDecay = {},
            Name = "99Ru",
            NaturalAbundance = 12.76,
        },
        {
            HalfLife = {},
            Mass = 99.904,
            MassNumber = 100,
            ModeOfDecay = {},
            Name = "100Ru",
            NaturalAbundance = 12.6,
        },
        {
            HalfLife = {},
            Mass = 100.906,
            MassNumber = 101,
            ModeOfDecay = {},
            Name = "101Ru",
            NaturalAbundance = 17.06,
        },
        {
            HalfLife = {},
            Mass = 101.904,
            MassNumber = 102,
            ModeOfDecay = {},
            Name = "102Ru",
            NaturalAbundance = 31.55,
        },
        {
            HalfLife = {},
            Mass = 103.905,
            MassNumber = 104,
            ModeOfDecay = {},
            Name = "104Ru",
            NaturalAbundance = 18.62,
        },
    },
    KeyIsotopes = {
        {
            HalfLife = {},
            Mass = 100.906,
            MassNumber = 101,
            ModeOfDecay = {},
            Name = "101Ru",
            NaturalAbundance = 17.06,
        },
        {
            HalfLife = {},
            Mass = 101.904,
            MassNumber = 102,
            ModeOfDecay = {},
            Name = "102Ru",
            NaturalAbundance = 31.55,
        },
        {
            HalfLife = {},
            Mass = 103.905,
            MassNumber = 104,
            ModeOfDecay = {},
            Name = "104Ru",
            NaturalAbundance = 18.62,
        },
    },
    MeltingPointK = 2606,
    Name = "Ruthenium",
    NeutronsApprox = 57,
    Period = 5,
    Protons = 44,
    RepresentativeMassNumber = 101,
    StateAt293K = "Solid",
    Symbol = "Ru",
    Uses = "Many new uses are emerging for ruthenium. Most is used in the electronics industry for chip resistors and electrical contacts. Ruthenium oxide is used in the chemical industry to coat the anodes of electrochemical cells for chlorine production. Ruthenium is also used in catalysts for ammonia and acetic acid production. Ruthenium compounds can be used in solar cells, which turn light energy into electrical energy.\
\
Ruthenium is one of the most effective hardeners for platinum and palladium, and is alloyed with these metals to make electrical contacts for severe wear resistance. It is used in some jewellery as an alloy with platinum.",
}

return Ruthenium
