local Thulium = {
    Name = "Thulium",
    Symbol = "Tm",
    AtomicNumber = 69,
    AtomicMass = 168.93,
    RepresentativeMassNumber = 169,
    Protons = 69,
    Electrons = 69,
    NeutronsApprox = 100,
    Category = "Lanthanoid",
    Group = "Lanthanides",
    Period = 6,
    Block = "f",
    StateAt293K = "Solid",
    Density = 9.32,
    Appearance = {
        "A bright, silvery metal.",
    },
    Uses = {
        "When irradiated in a nuclear reactor, thulium produces an isotope that emits x-rays. A 'button' of this isotope is used to make a lightweight, portable x-ray machine for medical use. Thulium is used in lasers with surgical applications.",
    },
    ElectronConfiguration = "[Xe] 4f136s2",
    EnergyLevels = {
        2,
        8,
        18,
        31,
        8,
        2,
    },
    AtomicRadius = 2.27,
    CovalentRadius = 1.77,
    ElectronAffinity = 99.283,
    Electronegativity = 1.25,
    IonizationEnergies = {
        596.695,
        1162.65,
        2284.77,
        4119.9,
    },
    MeltingPointK = 1818,
    BoilingPointK = 2223,
    KeyIsotopes = {
        {
            Name = "169Tm",
            HalfLife = {},
            Mass = 168.934,
            MassNumber = 169,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
    Isotopes = {
        {
            Name = "169Tm",
            HalfLife = {},
            Mass = 168.934,
            MassNumber = 169,
            ModeOfDecay = {},
            NaturalAbundance = 100,
        },
    },
}

return Thulium
